polygeist/test/polygeist-opt/bareptrlowering.mlir

```diff
@@ -461,7 +461,7 @@ func.func private @subindexop_memref_same_dim(%arg0: memref<4x4xf32>, %arg1: ind
 // CHECK-SAME:                                        %[[VAL_0:.*]]: !llvm.ptr) -> !llvm.ptr
 // CHECK-NEXT:      %[[VAL_1:.*]] = llvm.mlir.constant(0 : index) : i64
 // CHECK-NEXT:      %[[VAL_2:.*]] = llvm.mlir.constant(0 : i64) : i64
-// CHECK-NEXT:      %[[VAL_3:.*]] = llvm.getelementptr %[[VAL_0]]{{\[}}%[[VAL_2]], %[[VAL_1]]] : (!llvm.ptr, i64, i64) -> !llvm.ptr, f32
+// CHECK-NEXT:      %[[VAL_3:.*]] = llvm.getelementptr %[[VAL_0]]{{\[}}%[[VAL_2]], 0] : (!llvm.ptr, i64) -> !llvm.ptr, !llvm.struct<(f32)>
```

Comment:
VAL_1 is no longer used, so it would be nice if the compiler was not generating the operation (at line 462). If that is difficult then I would adjust the test to remove the CHECK-NEXT at line 462. Same comment for similar patterns in this file.

Reply:
These constants are removed by the canonicalization pass. Removing the `CHECK-NEXT` line for them would mean we might miss other operations that are not supposed to be there (because the line for `VAL_2` would need to be `CHECK` instead of `CHECK-NEXT`).